internal/services/signalr/signalr_service_resource.go

```diff
@@ -468,6 +469,45 @@ func signalRFeature(featureFlag signalr.FeatureFlags, value string) signalr.Sign
 	}
 }
 
+func expandUpstreamAuth(input []interface{}) *signalr.UpstreamAuthSettings {
+	if len(input) == 0 {
+		return nil
+	}
+
+	result := signalr.UpstreamAuthSettings{}
+
+	inputRaw := input[0].(map[string]interface{})
+
+	authType := signalr.UpstreamAuthType(inputRaw["type"].(string))
+	result.Type = &authType
```

Comment:
given the block being present means we can rely on the `user_assigned_identity_id` field being set, we can make this:

```suggestion
	result := signalr.UpstreamAuthSettings{
		Type: signalr.UpstreamAuthTypeManagedIdentity,
	}

	inputRaw := input[0].(map[string]interface{})
```

Reply:
Thanks Tom for the suggestion, the managedIdentity id can be system assigned and user assigned, it doesn't restricted to user assigned only, so we may better keep the value as managed identity to avoid confusion?